verifier/detail_template.html

```diff
@@ -101,20 +101,28 @@
       // criteria.
       let fail_characterized_json;
 
-      fetch('./pass.json')
-    .then((response) => response.json())
-    .then((data) => {pass_json = data});
-      fetch('./failing_tests.json')
-    .then((response) => response.json())
-    .then((data) => {fail_json = data});
-      fetch('./test_errors.json')
-    .then((response) => error_json = response.json())
-    .then((data) => {error_json = data});
-      fetch('./unsupported.json')
-    .then((response) => unsupportd_json = response.json())
-    .then((data) => {unsupported_json = data});
-
-      // TODO: handle promises to refresh screen when data is loaded.
+      let p1 = fetch('./pass.json');
+      let p2 = fetch('./failing_tests.json');
+      let p3 = fetch('./test_errors.json');
+      let p4 = fetch('./unsupported.json');
+
+      // Synchronize all the data loading and charts / pagination
+      Promise.all([
+          p1.then((response) => response.json())
+              .then((data) => {pass_json = data}),
+          p2.then((response) => response.json())
+              .then((data) => {fail_json = data}),
+          p3.then((response) => response.json())
+              .then((data) => {error_json = data}),
+          p4.then((response) => response.json())
+              .then((data) => {unsupported_json = data}),
+          new Promise((resolve, reject) => {
+              $(document).ready(resolve);
+          })
+      ]).then(([p1, p2, p3, p4, _ready]) => {
+          google.charts.setOnLoadCallback(drawChart);
+          onloadFn();  // To call the pagination setup.
```

Comment:
In this file, you didn't remove the other call sites of these two functions. Please do so:

```
      // Set a callback to run when the Google Visualization API is loaded.
      google.charts.setOnLoadCallback(drawChart);
```

and

```
  <body onload=onloadFn()>
```